content/en-US/api/roundRect.md

```diff
@@ -0,0 +1,113 @@
+specifications: https://html.spec.whatwg.org/#dom-context-2d-roundrect
+browser_compatibility: api.CanvasRenderingContext2D.roundRect
+---
+
+
+## Description
+
+The `roundRect` method of the `CanvasRenderingContext2D` interface adds a rounded rectangle to the 
+current sub-path.
+
+## Syntax
+
+`CanvasRenderingContext2D.roundRect(x, y, width, height, radii)`
+
+### Parameters
+<dl>
+  <dt><em>Parameter</em></dt>
+  <dd>Definition</dd>
+  <dt><em>x</em></dt>
+  <dd>A double that represents the x-axis (horizontal) coordinate of the rectangle's starting point.</dd>
+
+  <dt><em>y</em></dt>
+  <dd>A double that represents the y-axis (vertical) coordinate of the rectangle's starting point.</dd>
+
+  <dt><em>width</em></dt>
+  <dd>A double that represents the rectangle's width. Positive value means the path is drawn
+  clockwise; and negative value meeans it's drawn counterclockwise and flips the rectangle
+  horizontally along `y`, that means the radius values that applied to the left corners are now
+  applied to the right.</dd>
+
+  <dt><em>height</em></dt>
+  <dd>A double that represents the rectangle's height. Positive value means the path is drawn
+  clockwise; and negative value meeans it's drawn counterclockwise and flips the rectangle
+  vertically along `x`, that means the radius values that applied to the top corners are now applied
+  to the bottom.</dd>
+
+  <dt><em>radii</em></dt>
+  <dd>Each value `r` in `radii`  could be a number or an object with `{x, y}` properties. If `r` is 
+  a number, the corresponding corner(s) are drawn as a cicurlar arc with radius `r`; if `r` is an
+  object, the correspondinf corner(s) are drawn as 
+  [elliptical arc](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/ellipse)
+  whose `radiusX` and `radiusY` are equal to `x` and `y`, respectively. `r`, `x` or `y` must be
+  non-negative:
+
+  * If `radii`'s size is 1, then arc created by `radii[0]` replaces all 4 corners of rectangle. 
+
+  * If `radii`'s size is 2, then the arc created by `radii[0]` replaces the upper left and lower
+  right corners of the rectangle; the arc created by `radii[1]` replaces the upper right and lower
+  left corners of rectangle.
+
+  * If `radii`'s size is 3, then the arc created by `radii[0]` replaces the upper left corner; the
+  arc created by `radii[1]` replaces the upper right and lower left corners; and the arc created by
+  `radii[2]` replaces the lower right corner of rectangle.
+
+
+  * If `radii`'s size is 4, then the arc createed by radii[0], radii[1], radii[2] and radii[3]
+  replaces upper left, upper right, lower right and lower left corner of rectangle, respectively.
+
+  * If `radii`'s size is not any of the listed values above, then roundRect aborts and returns an
+  RangeError.
+
+  Note that if the sum of the radii of two corners of the same edge is greater than the length of
+  that edge, all the `radii` of the rounded rectangle are scaled by a factor of length / (r1 + r2).
```

Comment:
Not quite true. If multiple edges are smaller than their respective radii, all radii are scaled by the ratio that corresponds to the edge that requires the most scaling. I think the text here explains it well (https://github.com/whatwg/html/pull/6765)

Reply:
corrected! I over simplified it.